Source/WebCore/style/AnchorPositionEvaluator.cpp

```diff
@@ -734,18 +736,33 @@ void AnchorPositionEvaluator::updateAnchorPositioningStatesAfterInterleavedLayou
 
     for (auto elementAndState : document.styleScope().anchorPositionedStates()) {
         auto& state = *elementAndState.value;
-        if (state.stage == AnchorPositionResolutionStage::Initial) {
+        if (state.stage == AnchorPositionResolutionStage::FindAnchors) {
             Ref element { elementAndState.key };
-            if (element->renderer())
+            if (CheckedPtr renderer = element->renderer()) {
                 state.anchorElements = findAnchorsForAnchorPositionedElement(element, state.anchorNames, anchorsForAnchorName);
-            state.stage = AnchorPositionResolutionStage::FoundAnchors;
+                if (isLayoutTimeAnchorPositioned(renderer->style()))
+                    renderer->setNeedsLayout();
```

Comment:
Maybe I'm missing something, but why do we invalidate layout here too when we've already invalidated in `Scope::invalidateForAnchorDependencies`?

Reply:
`invalidateForAnchorDependencies` is only triggered when something changes in the layout. In case of the first resolution, the anchor and anchored will be positioned (by the first interleaved layout) before we have connected the anchors so we need another layout for the anchored elements. Basically needed to get the first PASS in imported/w3c/web-platform-tests/css/css-anchor-position/anchor-center-offset-change.html

Not super pretty but work for now.